src/snowflake/snowpark/modin/plugin/_internal/type_utils.py

```diff
@@ -295,7 +297,7 @@ def column_astype(
     if to_dtype == np.object_:
         return to_variant(curr_col)
 
-    if from_sf_type == to_sf_type:
+    if from_sf_type == to_sf_type and not isinstance(to_sf_type, TimedeltaType):
```

Comment:
why we need to change this?

Reply:
Oh you're right if `from_sf_type` is `LongType()` and `to_sf_type()` is `TimedeltaType()` which inherits from `LongType`, the type is already correct so it doesn't matter is we short circuit. Will remove.